File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int longest = 0;
        int count = 0;
        unordered_set<int> sett;

        for(int i = 0; i < nums.size(); i++){
            sett.insert(nums[i]);
        }
        int x;
        for(auto it : sett){
            if(sett.find(it - 1) == sett.end()){
                count = 1;
                x = it;

                while(sett.find(x + 1) != sett.end()){
                    x = x + 1;
                    count++;
                }
            }
            longest = max(longest, count);
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        sett = set(nums)
        longest = 0
        count = 0
        for val in sett:
            if val - 1 not in sett:
                x = val
                count = 1

                while x + 1 in sett:
                    x = x + 1
                    count+=1
            
            longest = max(longest, count)
        return longest